src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,594 @@
+        return $this;
+    }
+
+    /**
+     * Get the field to use for the label of the option
+     *
+     * The default value of 'Title' will map to DataObject::getTitle() if a Title DB field does not exist
+     */
+    public function getLabelField(): string
+    {
+        return $this->labelField;
+    }
+
+    /**
+     * Set the field to use for the label of the option
+     */
+    public function setLabelField(string $labelField): static
+    {
+        $this->labelField = $labelField;
+        return $this;
+    }
+
+    public function getAttributes(): array
+    {
+        $name = $this->getName();
+        if ($this->getIsMultiple() && strpos($name, '[') === false) {
+            $name .= '[]';
+        }
+        return array_merge(
+            parent::getAttributes(),
+            [
+                'name' => $name,
+                'data-schema' => json_encode($this->getSchemaData()),
+            ]
+        );
+    }
+
+    /**
+     * Get a list of selected ID's
+     */
+    public function getValueArray(): array
+    {
+        $value = $this->Value();
+        if (empty($value)) {
+            return [];
+        }
+        if (is_array($value)) {
+            $arr = $value;
+            // Normalise FormBuilder values to be like Page EditForm values
+            //
+            // Page EditForm $values for non-multi field will be
+            // [
+            //   0 => '10',
+            // ];
+            // FormBuilder $values for non-multi field will be
+            // [
+            //   'label' => 'MyTitle15', 'value' => '10'
+            // ]
+            if (array_key_exists('value', $arr)) {
+                $arr = [$arr['value']];
+            }
+            // Page EditForm $values for multi will be
+            // [
+            //   0 => '10',
+            //   1 => '15'
+            // ];
+            // FormBuilder $values for multi will be
+            // [
+            //   0 => ['label' => '10', 'value' => 'MyTitle10', 'selected' => false],
+            //   1 => ['label' => '15', 'value' => 'MyTitle15', 'selected' => false]
+            // ];
+            if (array_key_exists(0, $arr) && is_array($arr[0]) && array_key_exists('value', $arr[0])) {
+                $arr = array_map(fn ($a) => $a['value'], $arr);
+            }
+            $arr = array_filter($arr);
+            return array_map(fn ($a) => (int) $a, $arr);
+        }
+        try {
+            // things stdClass cannot be cast to string
+            (string) $value;
+        } catch (Error $e) {
+            $value = 0;
+        }
+        if (ctype_digit((string) $value) && $value != 0) {
+            return [(int) $value];
+        }
+        if ($value instanceof SS_List) {
+            return array_filter($value->column('ID'));
+        }
+        if ($value instanceof DataObject && $value->exists()) {
+            return [$value->ID];
+        }
+        // Don't know what value is, handle gracefully
+        return [];
```

Comment:
Discarding the value doesn't seem like a graceful way to handle it - we should probably throw an exception instead?

Reply:
I considered that, though because this is a form field used in the CMS it could leave CMS editors in a situation where the CMS just breaks and there's nothing that can do about it.

Reply:
That makes sense.
In that case we should use `user_error($message, E_USER_WARNING)`, or get the logger via `Injector::inst()->get(LoggerInterface::class)` and log a warning.

Reply:
Done